scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2166 @@
+
+"""
+
+class AndroidToolError(Exception):
+    pass
+
+
+class AndroidGradlePluginRequirements(object):
+    """
+    This class manages the Android Gradle plugin requirements environment and represents each entry from the
+    ANDROID_GRADLE_PLUGIN_COMPATIBILITY_MAP.
+    """
+    def __init__(self,
+                 agp_version:str,
+                 gradle_version:str,
+                 sdk_build_tools_version:str,
+                 jdk_version:str,
+                 release_note_url:str):
+        self._agp_version = Version(agp_version)
+        self._gradle_ver = Version(gradle_version)
+        self._sdk_build_tools_version = Version(sdk_build_tools_version)
+        self._jdk_version = Version(jdk_version)
+        self._release_notes_url = release_note_url
+
+    @property
+    def gradle_ver(self) -> Version:
+        return self._gradle_ver
+
+    @property
+    def sdk_build_tools_version(self) -> Version:
+        return self._sdk_build_tools_version
+
+    @property
+    def version(self) -> Version:
+        return self._agp_version
+
+    def validate_java_version(self, java_version:str) -> None:
+        """
+        Validate a version of Java against the current Android Gradle Plugin. Raise a detailed exception if it doesn't meet the requirements.
+        @param java_version:    The version string reported by java (-version)
+        @return: None
+        """
+        java_version_check = Version(java_version)
+        if not java_version_check >= self._jdk_version:
+            raise AndroidToolError(f"The installed version of java ({java_version_check}) does not meet the minimum version of ({self._jdk_version}) "
+                                   f"which is required by the Android Gradle Plugin version ({self._agp_version}). Refer to the android gradle plugin "
+                                   f"release notes at {self._release_notes_url}")
+
+    def validate_gradle_version(self, gradle_version:str) -> None:
+        """
+        Validate a version of gradle against the current Android Gradle Plugin. Raise a a detailed exception if it doesnt meet the requirements.
+        @param java_version:    The version string reported by gradle. (--version)
+        @return: None
+        """
+        gradle_version_check = Version(gradle_version)
+        if not gradle_version_check >= self._gradle_ver:
+            raise AndroidToolError(f"The installed version of gradle ({gradle_version_check}) does not meet the minimum version of ({self._gradle_ver}) "
+                                   f"which is required by the Android Gradle Plugin version ({self._agp_version}). Refer to the android gradle plugin "
+                                   f"release notes at {self._release_notes_url}")
+
+# The ANDROID_GRADLE_PLUGIN_COMPATIBILITY_MAP manages the known android plugin known to O3DE and its compatibility requirements
+# Note: This map needs to be updated in conjunction with newer versions of the Android Gradle plugins.
```

Comment:
maybe we should log this message when the user tries to specify a newer gradle version

Reply:
Added logic to detect newer versions with better messaging